### ISSUE TITLE
No crawlers with Lifestyle mod

### ISSUE THREAD

**Issue by Bibibububobo** [theCrius/project-zomboid-random-zombies-full#11]
**Describe the bug**
Hi, I love your mod btw... Fine tuning the crowds brings more realism and game cant be boring now. Thank you for working on it. 
Now on the strange bug I just found: a surprising incompatibility with the Lifestyle mod (2997342681) about crawlers.

**To Reproduce**
Install your mod and Lifestyle mod. Have a Z distrib with some crawlers. They wont appear ingame.

**Expected behavior**
Crawlers should exist ingame when they are in the distrib. I can only guess animations from Lifestyle do override the status of crawlers your mod applies.

**Specification (please complete the following information):**
- Game Version: latest (Steam)
- Game mode: MP on dedicated server. Didnt try on SP.
- SandboxVars block for the mod: default one. Well it is more visible with a high crawler amount set, of course.

**Comment by theCrius** [comment]
The status of crawler is exactly that. I'll need to check on the source code of this "Lifestyle mod" to understand if it's overriding.

However you can do a quick test by checking if the issue persist with the order of mods being

```
[....]
Random Zombies - Day and Night
[...]
Lifestyle Mod
[...]
```

And the other way around. Theoretically if they are invoking the same functions, the crawlers should work for this mod.
If instead the other mod is acting on the animation only, I am afraid they are going to be just incompatible.

You can see how the crawlers are set (which functions) from here directly:

https://github.com/theCrius/project-zomboid-random-zombies-full/blob/main/Contents/mods/RandomZombiesDayAndNight/media/lua/server/rzf_zombiesManager.lua#L135

**Comment by Bibibububobo** [comment]
Quick update: I tried with only both mods installed (no others), then tried swapping the load order, with both runs on a fresh start on dedicated server. Twice the same interaction: no crawlers to be seen.

I wrote to the other dev on his own Github and this is his answer, FYI:
https://github.com/Angry362/Lifestyle-Mod-Project-Zomboid-/issues/5#issuecomment-1735565484

**Comment by theCrius** [comment]
That is really weird. Did you try with something like 100% crawlers to be sure you haven't been just unlucky?

I'm sorry to ask you all these tests but I'm in the middle of some personal and work stuff and I'd rather jump in to fix issues only if I'm extra sure something's wrong.

What makes me confused is that there hasn't been any relevant update in the game as far as I am aware and the other mod developer says that he has no issue running both.

If you are willing to test this further I would try also just running random zombies alone with 100% crawlers to verify that by itself everything works.

Can you also link the other mod steam page here so I can easily add it to my list and do some tests myself in case yours are inconclusive?

Cheers!

**Comment by Bibibububobo** [comment]
Well I am in the setup and fine-tuning of an heavy modded dedicated server, so I did not see it at first since I checked some other features brought by other mods I was trying. A few days or even weeks could have gone by since a potential update issue.

Now, onto the test:
>         Crawler_Day = 100.0,
>         -- Minimum=0.00 Maximum=100.00 Default=65.00
>         Shambler_Day = 0.0,
>         -- Minimum=0.00 Maximum=100.00 Default=25.00
>         FastShambler_Day = 0.0,
>         -- Minimum=0.00 Maximum=100.00 Default=5.00
>         Sprinter_Day = 0.0,
>         -- Minimum=0.00 Maximum=100.00 Default=30.00
>         Crawler_Night = 100.0,
>         -- Minimum=0.00 Maximum=100.00 Default=25.00
>         Shambler_Night = 0.0,
>         -- Minimum=0.00 Maximum=100.00 Default=65.00
>         FastShambler_Night = 0.0,
>         -- Minimum=0.00 Maximum=100.00 Default=5.00
>         Sprinter_Night = 0.0,
>         -- Minimum=0.00 Maximum=100.00 Default=30.00

Let's try with and without Lifestyle:

> **Mods=RandomZombiesFull;Lifestyle;**
> WorkshopItems=2997342681;2888099799;

![steamuserimages-a akamaihd](https://github.com/theCrius/project-zomboid-random-zombies-full/assets/134651982/c2f1141e-6901-47f7-819a-8f776e3b8fdf)
(No crawlers...)


> **Mods=RandomZombiesFull;**
> WorkshopItems=2888099799;

![steamuserimages-a akamaihd2](https://github.com/theCrius/project-zomboid-random-zombies-full/assets/134651982/585e5691-280f-4c77-9bfb-e513478782cf)
(Crawlers everywhere, must be a fun run to try...)


Strange thing is, if both mods are active, when the zombies come into view, some may be on the ground (face up, not down), but they get up pretty quickly, and they do stay up. Not sure if that helps, but here is an example in this super-potato video:
https://drive.google.com/file/d/1k3S3i10rVPSZfd9AS1M2EdXCXPtLfRLY/view

The other Steam mod is 'Lifestyle' and can be found here:
https://steamcommunity.com/sharedfiles/filedetails/?id=2997342681

Many thanks for your help.

**Comment by theCrius** [comment]
Thank you for testing this.

The behaviour with just the random zombies is what is expected.
I don't know why the dev from the other mod is saying he doesn't have any issue.

If you want to make a last try, you can check the original mod I took inspiration and some of the original code from, Random Zombies by Belette: https://steamcommunity.com/sharedfiles/filedetails/?id=2818577583

And see if it happens as well with her mod.
If it doesn't, then there must have been some change that I missed and I'll check for differences over there. It could help fix this in a rather quick way (unless some hack have been needed).

**Comment by Bibibububobo** [comment]
Well I am sorry but I bear some bad news... or good ones, if we talk about finding the culprit  :)

100% crawlers with Random Zombies (belette) only:
![20230927225115_1](https://github.com/theCrius/project-zomboid-random-zombies-full/assets/134651982/695d7fe2-82d9-4d56-8270-4f3f1f7b9d18)

With RZ + Lifestyle:
![20230927225643_1](https://github.com/theCrius/project-zomboid-random-zombies-full/assets/134651982/983a2f94-fced-42a9-af4a-5e4eed9c42e0)

**Comment by theCrius** [comment]
That's good news then. I never touched the code that made the crawlers since I used belette's.
I'll check what changed and should have a fix soon enough.

I'll give it a look over the weekend. Thank you for the tests :)

**Comment by Bibibububobo** [comment]
That's great, thanks to you for diving into the code. If you need anymore testing, I can help again!

**Comment by Bibibububobo** [comment]
Hope you can have a look this weekend!   🤞

**Comment by theCrius** [comment]
Yeah, last weekend passed over with some family troubles... and this one I have to undergo a surgical operation. Fun times.

**Comment by Bibibububobo** [comment]
Did you find some time to look for a fix? Still want to play with both mods with my friends  🤐

**Comment by Angry362** [comment]
I hope you're having a good recovery my dude! ❤

**Comment by theCrius** [comment]
Hey, just letting you know I've not forgotten but things are simply getting more and more complicated. When personal life gets better I'll get back into this. If you meet or get to lean how to contribute, help is always welcome :)
